Firmware/stepper.cpp

```diff
@@ -230,7 +230,9 @@ bool endstops_hit_on_purpose()
 bool endstop_z_hit_on_purpose()
 {
   bool hit = endstop_hit & _BV(Z_AXIS);
+CRITICAL_SECTION_START;
```

Comment:
Nope, I'm strictly against it
The endstop_hit* flags must remain as standalone variables due to being used in time critical code. Any additional instructions are to be avoided.

Reply:
explained, closing this comment